src/components/Viewport.ts

```diff
@@ -347,6 +388,12 @@ export default class Viewport {
     return estimatedPosition;
   }
 
+  public addVisiblePanel(panel: Panel): void {
+    if (findIndex(this.visiblePanels, visiblePanel => visiblePanel === panel) < 0) {
```

Comment:
@WoodNeck
Here. `findIndex` It's like a overlapping section.

Reply:
You mean, like adding `private getVisibleIndex(panel: Panel)`? I think that's ok.